File: PBKit/PBLogger.m
//
//  PBLogger.m
//  PBKit
//
//  Created by pebble on 2017/04/02.
//  Copyright © 2017年 pebble8888. All rights reserved.
//

#import <Foundation/Foundation.h>

void Logf(NSString* fmt, ...)
{
    NSFileManager* fm = NSFileManager.defaultManager;
    NSString* path = @"/tmp/pbc.log";
    if (![fm fileExistsAtPath:path]) {
        [fm createFileAtPath:path contents: nil attributes: nil];
    }
    NSFileHandle* fh = [NSFileHandle fileHandleForWritingAtPath: path];
    if (fh != nil){
        va_list args;
        va_start(args, fmt);
        NSString* s = [[NSString alloc] initWithFormat:fmt arguments:args];
        va_end(args);
        
        NSData* data = [s dataUsingEncoding:NSUTF8StringEncoding];
        if (data != nil){
            [fh seekToEndOfFile];
            [fh writeData:data];
            [fh synchronizeFile];
        }
        [fh closeFile];
    }
}

File: PBKit/PBLogger.swift
//
//  PBLogger.swift
//
//  Created by pebble8888 on 2016/03/28.
//  Copyright © 2016年 pebble8888. All rights reserved.
//

import Foundation

public func DEBUG_LOG(_ logMessage: String, functionName: String = #function, line: Int = #line) {
    NSLog("\(functionName) Line:[\(line)] \(logMessage)")
}

public func debug_log(_ logMessage: String, functionName: String = #function, line: Int = #line) {
    let fm = FileManager.default
    let path = "/tmp/pb.log"
    if !fm.fileExists(atPath:path) {
        fm.createFile(atPath:path, contents: nil, attributes: nil)
    }
    if let fh = FileHandle(forWritingAtPath: path) {
        let date = Date()
        let s = "\(date.standardRepresentation) \(functionName) Line:\(line) \(logMessage)\n"
        if let data = s.data(using: .utf8) {
            fh.seekToEndOfFile()
            fh.write(data)
            fh.synchronizeFile()
        }
        fh.closeFile()
    }
}
